src/libsyntax/ast.rs

```diff
@@ -536,8 +554,14 @@ pub struct Pat {
 }
 
 #[derive(Clone, PartialEq, Eq, RustcEncodable, RustcDecodable, Hash, Debug)]
+/// A single field in a struct pattern
+///
+/// For patterns like `Foo {x, y, z}`, `pat` and `ident` point to the same identifier
```

Comment:
Maybe "For patterns like `x` in `Foo { x, y, z }` ..." just to be completely unambiguous.